scribble-lib/scribble/private/manual-style.rkt

```diff
@@ -219,26 +219,55 @@
         (cond
           [(string? i)
            (cond
-             [(regexp-match #px"^(.*)_([a-zA-Z0-9]+)(.*)$" i)
+             [(regexp-match #px"^(.*)\\^\\{(.*)\\}_\\{(.*)\\}(.*)$" i)
               => (lambda (m)
-                   (append (loop (cadr m))
-                           (list (make-element 'subscript
-                                               (loop (caddr m))))
-                           (loop (cadddr m))))]
-             [(regexp-match #px"^(.*)\\^([a-zA-Z0-9]+)(.*)$" i)
+                   (let* ([m (cdr m)] [s1 (car m)]
```

Comment:
Using `match` to extract `s1`, ... would be better.

Reply:
Note that `match` already supports a `pregexp` pattern, so you can replace the entire `cond` with a `match` on `i`.